File: formats/gzip/compressor/java/Inflate.java
import java.io.*;
import java.text.*;
import java.util.zip.*;

class Inflate
{
	static public void main(String argv[])
	{
		try 
		{
			ByteArrayOutputStream packed = new ByteArrayOutputStream();
			
			// Read data from standard input into memory
			byte[] chunk = new byte[102400];
			int n = System.in.read(chunk, 0, chunk.length);
			while (n > 0)
			{
				packed.write(chunk, 0, n);
				n = System.in.read(chunk, 0, chunk.length);
			}
			
			byte[] unpacked = new com.jbase.framework.io.GZipCompressor().uncompress(packed.toByteArray());

			System.out.write(unpacked, 0, unpacked.length);
		} 
		catch (java.io.IOException e)
		{
		}
	}
}

File: formats/gzip/compressor/cs/inflate.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace compressortest
{
	class Inflate
	{
		static void Main(string[] args)
		{
			MemoryStream raw = new MemoryStream();

			Stream stdin = System.Console.OpenStandardInput();
			byte[] chunk = new byte[10240];

			int n = stdin.Read(chunk, 0, chunk.Length);
			while (n > 0)
			{
				raw.Write(chunk, 0, n);
				n = stdin.Read(chunk, 0, chunk.Length);
			}

			byte[] unpacked = new com.jbase.framework.io.GZipCompressor().uncompress(raw.ToArray());

			System.Console.OpenStandardOutput().Write(unpacked, 0, unpacked.Length);
		}
	}
}
